pkg/provider/pods.go

```diff
@@ -432,7 +433,7 @@ func (p *Pod) IsUsingSRIOV() (bool, error) {
 }
 
 // IsUsingSRIOVWithMTU returns true if any of the pod's interfaces is a sriov one with MTU set.
-func (p *Pod) IsUsingSRIOVWithMTU() (bool, error) {
+func (p *Pod) IsUsingSRIOVWithMTU(env *TestEnvironment) (bool, error) {
```

Comment:
I would say that, in this function, we can keep the current code we have (this is, to check the NetworkAttachmentDefinition and detect if the sriov plugin is present and the mtu is defined), and if not finding anything, then I'd do what you're adding below, which is to see if there are SriovNetwork resources, and then check in the SriovNetworkNodePolicy.
So, I think it is fine to do both checks, first checking NAD as you do right now, secondly to check SRIOV resources if nothing was found in NAD.

Reply:
There's still something in this function that is not working when using a workload with NAD and SRIOV resources, see https://github.com/redhat-best-practices-for-k8s/certsuite/pull/2527#issuecomment-2434958388

Maybe it could be easier if we follow the approach proposed here? https://github.com/redhat-best-practices-for-k8s/certsuite/pull/2527#issuecomment-2431759789, directly using the pod network annotations?